__tests__/frontend-proxy.ts

```diff
@@ -0,0 +1,215 @@
+            headers: { 'X-Header': 'foo' },
+          })
+
+          expect(await response.text()).toBe('header-shown')
+        }
+      )
+
+      function checkRequestData(req: Request) {
+        const res =
+          req.headers.get('X-Header') === 'foo'
+            ? new Response('header-shown')
+            : new Response('no-header')
+
+        return new Promise<Response>((resolve) => resolve(res))
+      }
+    })
+  })
+
+  describe('transfers response meta data from backend', () => {
+    test.each([true, false])('useFrontend=%p', async (useFrontend) => {
+      const targetResponse = new Response('', {
+        headers: { 'X-Header': 'bar' },
+      })
+
+      await withMockedFetch([createApiResponse(), targetResponse], async () => {
+        const cookie = `useFrontend20=${useFrontend};`
+        const res = await handleUrl(testUrl, 0.2, cookie)
+
+        expect(res.headers.get('X-Header')).toBe('bar')
+      })
+    })
+  })
+
+  test('requests to /enable-frontend enable use of frontend', async () => {
+    const url = 'https://de.serlo.org/enable-frontend'
+    const res = await handleUrl(url, 0.15)
+
+    hasOkStatus(res)
+    expect(res.headers.get('Set-Cookie')).toBe('useFrontend15=true; path=/')
+    expect(await res.text()).toBe('Enable frontend')
+  })
+
+  test('requests to /disable-frontend disable use of frontend', async () => {
+    const url = 'https://de.serlo.org/disable-frontend'
+    const res = await handleUrl(url, 0.15)
+
+    hasOkStatus(res)
+    expect(res.headers.get('Set-Cookie')).toBe('useFrontend15=false; path=/')
+    expect(await res.text()).toBe('Disable frontend')
+  })
+})
+
+test('createApiQuery()', () => {
+  expect(createApiQuery('/math')).toBe(
+    '{ uuid(alias: { instance: de, path: "/math" }) { __typename } }'
+  )
+  expect(createApiQuery('hello')).toBe(
+    '{ uuid(alias: { instance: de, path: "/hello" }) { __typename } }'
+  )
+  expect(createApiQuery('/266/')).toBe(
+    '{ uuid(alias: { instance: de, path: "/266/" }) { __typename } }'
+  )
+  expect(createApiQuery('/266')).toBe('{ uuid(id: 266) { __typename } }')
+  expect(createApiQuery('/874329')).toBe('{ uuid(id: 874329) { __typename } }')
+})
+
+async function handleUrl(
+  url: string,
+  probability = 0.1,
+  cookie?: string,
+  reqInit?: RequestInit
+) {
+  const req = new Request(url, reqInit)
+
+  if (cookie !== undefined) req.headers.set('Cookie', cookie)
+
+  return (await handleRequest(req, probability, TEST_ALLOWED_TYPES)) as Response
+}
+
+export function createApiResponse(typename = TEST_ALLOWED_TYPES[0]) {
+  return createJsonResponse({ data: { uuid: { __typename: typename } } })
+}
+
+function createApiErrorResponse() {
+  return createJsonResponse({
+    errors: [{ message: 'error' }],
+    data: { uuid: null },
+  })
+}
+
+function checkFrontendRequest(req: Request) {
+  const res =
+    new URL(req.url).hostname === FRONTEND_DOMAIN
+      ? new Response('frontend')
+      : new Response('no-frontend')
```

Comment:
Not really sure what this is supposed to do. Is this a stub for the actual response that might either from serlo.org or frontend?